tradeexecutor/strategy/pandas_trader/position_manager.py

```diff
@@ -858,16 +859,87 @@ def open_credit_supply_position_for_reserves(self, amount: USDollarAmount) -> Li
             List of trades that will open this credit position
         """
 
-        lending_pool_identifier = self.strategy_universe.get_credit_supply_pair()
-        state = self.state
+        lending_reserve_identifier = self.strategy_universe.get_credit_supply_pair()
 
-        credit_supply_position, trade, _ = state.supply_credit(
+        _, trade, _ = self.state.supply_credit(
             self.timestamp,
-            lending_pool_identifier,
+            lending_reserve_identifier,
             collateral_quantity=Decimal(amount),
             trade_type=TradeType.rebalance,
             reserve_currency=self.strategy_universe.get_reserve_asset(),
             collateral_asset_price=1.0,
         )
 
         return [trade]
+    
+    def open_short(
+        self,
+        pair: Union[DEXPair, TradingPairIdentifier],
+        value: USDollarAmount,
+        *,
+        leverage: float = 1.0,
+        take_profit_pct: float | None = None,
+        stop_loss_pct: float | None = None,
+    ) -> list[TradeExecution]:
+        """Open a short position.
+
+        :param pair:
+            Trading pair where we take the position
+
+        :param value:
+            How large position to open, in US dollar terms
+
+        :param leverage:
+            Leverage level to use for the short position
+
+        :param take_profit_pct:
+            If set, set the position take profit relative
+            to the current market price.
+            1.0 is the current market price.
+            If asset opening price is $1000, take_profit_pct=1.05
+            will sell the asset when price reaches $1050.
+
+        :param stop_loss_pct:
+            If set, set the position to trigger stop loss relative to
+            the current market price.
+            1.0 is the current market price.
+            If asset opening price is $1000, stop_loss_pct=0.95
+            will sell the asset when price reaches 950.
+
+        :return:
+            List of trades that will open this credit position
+        """
+
+        if isinstance(pair, DEXPair):
+            executor_pair = translate_trading_pair(pair)
+        else:
+            executor_pair = pair
+
+        shorting_pair = self.strategy_universe.get_shorting_pair(executor_pair)
+
+        if type(value) == float:
+            value = Decimal(value)
+
+        # TODO: verify calculation here
+        price_structure = self.pricing_model.get_sell_price(self.timestamp, executor_pair, value)
+        borrowed_quantity = value * leverage / Decimal(price_structure.price)
```

Comment:
We have a function to calculate leverage here 

https://github.com/tradingstrategy-ai/trade-executor/blob/d885412d23283bcaa171da1106674bd1ed11f54e/tradeexecutor/state/loan.py#L271

It makes it simpler if we need to do special adjustments for these formulas later